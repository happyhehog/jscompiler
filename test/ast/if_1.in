if (a === 3) {
    hello();
}