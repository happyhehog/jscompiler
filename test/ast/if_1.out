ProgramNode (1:0, 3:0)
    IfStatementNode (1:0, 3:0)
    --> test expression:
        BinaryExpressionNode (1:4, 1:10) { operator: === }
        --> left operand:
            IdentifierNode (1:4, 1:4) : { name: a }
        --> right operand:
            LiteralNode (1:10, 1:10) : { value: 3 }
    --> consequent:
        BlockStatementNode (1:13, 3:0)
            CallExpressionNode (2:4, 2:10)
            --> callable:
                IdentifierNode (2:4, 2:4) : { name: hello }
